### DISCUSSION TITLE
Annotation for Mapping DAO result

### DISCUSSION THREAD

**Discussion by TheRealRyGuy** [jdbi/jdbi#2346] [Q&A] [answered]
Hello! 
In a short version, in an `@Repository`, I have a query running to collect a Set via a constructor mapper. However, I am looking for an annotation to see if I can further map this query into `Map<Enum Key, Set<Object>>`, mapping by an enum stored as an ordinal 
Can't try `array_agg` because JDBI is processing through a constructor data, so essentially I think I'm looking for an equivalent annotation to `ResultBearing#mapTo` that still allows me to access a db row. 
- Using jdbi-bom 3.38.0

**Comment by hgschmie** [comment]
Hi,

(I mentally filled in some blanks. `@Repository` is some spring thing I assume). So you return a `Set<SomeBean>` from a dao method or do you put this together programmatically? 

For a complex collection, you will most likely need to use a collector as described in the docs: https://jdbi.org/#_collectors

**Reply by TheRealRyGuy** [reply]
Trying to return a `Map<Enum, Set<non primitive Object>>` from a dao method - I need a complex collector, but was wondering if there was an annotation based way to do this. 
`ColumnMapper`s won't work cause I need to the full result set, and it doesn't look like there's anything implemented for it. I can take a stab at implementing an `@RegisterCollector` kind of deal, but thought it was worth asking beforehand

**Comment by stevenschlansker** [comment]
If your desired result doesn't "look like" a normal collection, it could also be helpful to check out the `reduceRows` or `reduceResultSet` - or even `scanResultSet`, which get you closer to JDBC and more control over how the result is constructed.

https://jdbi.org/apidocs/org/jdbi/v3/core/result/ResultBearing.html

**Comment by TheRealRyGuy** [comment, marked as answer]
Solved my own question https://github.com/jdbi/jdbi/pull/2357
